src/cascadia/QueryExtension/ILMProvider.idl

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Query.Extension
+{
+    interface ILMProvider
+    {
+        // chat related functions
+        void ClearMessageHistory();
+        void SetSystemPrompt(String systemPrompt);
+        void SetContext(IContext context);
+
+        Windows.Foundation.IAsyncOperation<IResponse> GetResponseAsync(String userPrompt);
+
+        // auth related functions
+        void SetAuthentication(Windows.Foundation.Collections.ValueSet authValues);
+        event Windows.Foundation.TypedEventHandler<ILMProvider, String> AuthChanged;
```

Comment:
These don't look consistent.

I would expect that if I catch an `AuthChanged` event, I could save the value it gave me (second argument) and give it right back to `SetAuthentication` later. Effectively, I expect `AuthChanged` to give me an opaque blob of something which `SetAuthentication` knows what to do with.